### PR TITLE
KerrSchild analytic solution now works for different frames.

### PR DESCRIPTION
**Pull request by markscheel**
## Proposed changes

KerrSchild now can be used for multiple frames.

Its member functions are now instantiated for Frame::Grid and Frame::Inertial.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp [hunk 1]
```diff
@@ -246,34 +248,33 @@ class KerrSchild : public MarkAsAnalyticSolution {
...
-      gr::Tags::InverseSpatialMetric<volume_dim, Frame::Inertial, DataType>>;
+      gr::Tags::ExtrinsicCurvature<volume_dim, Frame, DataType>,
+      gr::Tags::InverseSpatialMetric<volume_dim, Frame, DataType>>;
 
   template <typename DataType, typename... Tags>
-  tuples::TaggedTuple<Tags...> variables(const tnsr::I<DataType, volume_dim>& x,
-                                         double /*t*/,
-                                         tmpl::list<Tags...> /*meta*/) const
-      noexcept {
+  tuples::TaggedTuple<Tags...> variables(
```
Comment (nilsvu) [inline src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp:272]: I believe you only need to template this function on `Frame`, plus all the things that are currently templated on `DataType` in this class. Then the public interface can remain unchanged. Do you think that'll work?

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp (hunk 1 above)
Comment (markscheel) [reply src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp:272]: Yes, this works!  The changes to KerrSchild are more complicated, but the only things that need to change are KerrSchild and the test.

**Comment by markscheel** [comment]
Made changes and rebased.  Since the changes were so extensive, I didn't bother with fixup commits and I just re-pushed the new version.

**Review by nilsdeppe — DISMISSED** [review]

**Comment by markscheel** [comment]
Fixed test that was failing on older versions of gcc (variable was set outside an 'if constexpr' but used only inside one branch of 'if constexpr') and squashed.

**Review by nilsdeppe — APPROVED** [review]

**Review by nilsvu — APPROVED** [review]

**Comment by nilsvu** [comment]
@markscheel just like in #3005 I deleted the "do nothing" upgrade instructions because I'm preparing the release. Please add upgrade instructions if you think they would be useful here.

**Pull request merged**